src/library/VUMeter.js

```diff
@@ -0,0 +1,93 @@
+import FIFOArray from './FIFOarray.js';
+
+/**
+ * This is a VUMeter Library for creating VUMeter. It requires the 
+ * ID of canvas element where the waveform will be rendered and min
+ * decibel of for VU-meter as initialization input. minDecibel define
+ * the range of vu-meter to consider silent. When the waveform is created,
+ * it requires the analyserNode as real-time audio input data for 
+ * calculating the correct vm-meter library.
+ * @param {String} canvasId The HTML id where waveform is located.
+ * @param {Number} minDecibel The minimum lower bond of decible
+ *     If the decible is lower than this value, it cosider as slient.
+ * @param {AnalyserNode} analyserNode The analysis node
+ *     which connect with the audio context.
+ */
+class VUMeter {
+  constructor(canvasId, minDecible, analyserNode) {
+    this.canvas_ = document.querySelector(canvasId);
+    this.canvasContext_ = this.canvas_.getContext('2d');
+    this.width_ = this.canvas_.width;
+    this.height_ = this.canvas_.height;
+    this.currentX_ = 0;
+    this.previousY_ = this.height / 2;
+    // Due to the calculation and graph convinience,
+    // we set it to positive
+    this.minDisplayDecible_ = Math.abs(minDecible);
+    this.fifoArray_ = new FIFOArray(8);
+    this.analyser_ = analyserNode;
+    // This is the minimum fftSize which we are able to have.
+    // By using size 32, we can collect the most accurate data.
+    this.analyser_.fftSize = 32;
+    this.bufferLength_ = analyserNode.frequencyBinCount;
+    this.dataArray_ = new Uint8Array(this.bufferLength_);
+  }
+
+  /**
+   * This is the create function for VUMeter
+   */
+  draw() {
+    this.analyser_.getByteTimeDomainData(this.dataArray_);
+    
+    let formatDataArray = Array(this.dataArray_.length);
```

Comment:
Create an array up front in the constructor and reuse it in the draw() function.